build.gradle.kts

```diff
@@ -55,73 +47,30 @@ kotlin {
         browser()
         nodejs()
     }
-    wasm {
+    @OptIn(ExperimentalWasmDsl::class)
+    wasmJs {
         browser()
+        nodejs()
         binaries.executable()
     }
 
-    sourceSets {
-        val commonMain by getting
-        val commonTest by getting {
-            dependencies {
-                implementation(kotlin("test"))
+    @Suppress("OPT_IN_USAGE")
+    applyDefaultHierarchyTemplate {
+        common {
+            group("jsAndWasmJs") {
+                withJs()
+                withWasm()
             }
         }
+    }
 
-        val jsWasmMain by creating {
-            dependsOn(commonMain)
-        }
-        val jsWasmTest by creating {
-            dependsOn(commonTest)
-        }
-        val jsMain by getting {
-            dependsOn(jsWasmMain)
-        }
-        val jsTest by getting {
-            dependsOn(jsWasmTest)
-        }
-        val wasmMain by getting {
-            dependsOn(jsWasmMain)
-        }
-        val wasmTest by getting {
-            dependsOn(jsWasmTest)
-        }
-
-        val nativeCommonMain by creating {
-            dependsOn(commonMain)
-        }
-        val nativeCommonTest by creating {
-            dependsOn(commonTest)
-        }
-
-        val nativeDarwinMain by creating {
-            dependsOn(nativeCommonMain)
-        }
-        val nativeLinuxMain by creating {
-            dependsOn(nativeCommonMain)
-        }
-        val mingwMain by getting {
-            dependsOn(nativeCommonMain)
-        }
-
-        targets.withType<KotlinNativeTarget>().all {
-            val mainSourceSet = compilations.getByName("main").defaultSourceSet
-            val testSourceSet = compilations.getByName("test").defaultSourceSet
-
-            mainSourceSet.dependsOn(nativeCommonMain)
-            testSourceSet.dependsOn(nativeCommonTest)
-
-            when {
-                konanTarget.family == MINGW -> mainSourceSet.dependsOn(mingwMain)
-                konanTarget.family == LINUX || konanTarget.family == ANDROID -> mainSourceSet.dependsOn(
-                    nativeLinuxMain
-                )
-
-                konanTarget.family.isAppleFamily -> mainSourceSet.dependsOn(nativeDarwinMain)
-                else -> mainSourceSet.dependsOn(nativeCommonMain)
-            }
-        }
+    sourceSets.commonTest.dependencies {
+        implementation(kotlin("test"))
     }
 }
 
-apply(plugin = "com.vanniktech.maven.publish")
+// https://github.com/Kotlin/kotlin-wasm-examples/commit/701a051d6ee869abcabebff702b3ccd98d51c38d
+rootProject.the<NodeJsRootExtension>().apply {
+    nodeVersion = "21.0.0-v8-canary202309143a48826a08"
+    nodeDownloadBaseUrl = "https://nodejs.org/download/v8-canary"
+}
```

Comment:
Do you know if this is required for libraries? I thought they were only needed for the module that would produce the final output.

Reply:
I think I needed this for running tests. For some reason, they were failing. I'll double check

Reply:
Ktor has it and also had a comment related to why it's there
https://github.com/ktorio/ktor/blob/6e52d76177a823b3c667989483e9bce35b04f53c/build.gradle.kts#L229

I'll still check if the project works without it. In that case, I'll remove it

Reply:
My recollection was right. Test failure without the above block
<img width="1286" alt="Screenshot 2024-03-20 at 11 46 18 PM" src="https://github.com/touchlab/TestHelp/assets/2466073/d2a0a00e-a50d-4a2c-9f6a-9f92fde92943">